toolchain/driver/compile_subcommand.cpp

```diff
@@ -778,26 +919,25 @@ auto CompilationUnit::PostCompile() -> void {
 }
 
 auto CompilationUnit::RunCodeGenHelper() -> bool {
-  std::optional<CodeGen> codegen =
-      CodeGen::Make(module_.get(), options_->codegen_options.target, consumer_);
-  if (!codegen) {
-    return false;
-  }
+  CARBON_CHECK(module_, "Must call RunLower first");
```

Comment:
```suggestion
  CARBON_CHECK(module_, "Must call PostLower first");
```
Maybe? Should PostLower have some bool sideeffect we could check here?

Reply:
I don't know how valuable these CHECKs are at all; I only added them for consistency with the existing CHECKs. But we're not actually depending on the side-effects of `PostLower` here, only those of `RunLower`, and we're only checking that `RunLower` was run.

If we want something better here, perhaps we could add general tracking of the most recent phase and ensure we run them in the right order?

Reply:
I would prefer that, yeah. A nice enum would be much more clear

Reply:
I think this is best done in a separate PR. Would that work for you?

Reply:
Yes that would be great